jujuclient/interface.go

```diff
@@ -101,6 +101,8 @@ type AccountDetails struct {
 	// They are only set when using the MemStore implementation,
 	// and are used by embedded commands. The are not written to disk.
 	Macaroons []macaroon.Slice `yaml:"-"`
+
+	AccessToken string `yaml:"access-token,omitempty"`
```

Comment:
Needs a comment

Reply:
I thought we discussed adding a "login-type" or "login-provider" attribute and we'd use a factory type pattern to instantiate the login provider to use with the state (renamed to conn in juju 4.0 BTW)

Reply:
we did discuss it.. but, as my comment in the PR says, that is a much much larger change due to the many places where AccountDetails are used. And it doesn't give us much, since we did say that we want to use the same accounts.yaml file and be backwards compatible. Well this simple addition is backwards compatible, so a 2.9 client is still able to read the accoutns.yaml file..  And the change to use a factory type pattern should possibly be done in a followup to keep PRs small and focused